lib/service/servicecfg/config_test.go

```diff
@@ -694,3 +697,43 @@ func TestWebPublicAddr(t *testing.T) {
 		})
 	}
 }
+
+func TestSetLogLevel(t *testing.T) {
+	for _, test := range []struct {
+		logLevel            slog.Level
+		expectedLogrusLevel logrus.Level
+	}{
+		{
+			logLevel:            logutils.TraceLevel,
+			expectedLogrusLevel: logrus.TraceLevel,
+		},
+		{
+			logLevel:            slog.LevelDebug,
+			expectedLogrusLevel: logrus.DebugLevel,
+		},
+		{
+			logLevel:            slog.LevelInfo,
+			expectedLogrusLevel: logrus.InfoLevel,
+		},
+		{
+			logLevel:            slog.LevelWarn,
+			expectedLogrusLevel: logrus.WarnLevel,
+		},
+		{
+			logLevel:            slog.LevelError,
+			expectedLogrusLevel: logrus.ErrorLevel,
+		},
+	} {
+		t.Run(test.logLevel.String(), func(t *testing.T) {
+			c := MakeDefaultConfig()
+			require.NotNil(t, c)
+
+			c.SetLogLevel(test.logLevel)
```

Comment:
Will this overwrite the logging level used for running the test? 
https://github.com/gravitational/teleport/blob/a320e60ab5cab946e3a5c08a7b75b0946dc80d14/lib/utils/cli.go#L168-L171

Reply:
It probably will. We can switch the level back after the test is done. I've tested with more packages (not the entire test suite), and the log level was kept.

Reply:
I'll update the test not to use `ApplyDefaults` (which modifies the global logger). So we don't need to worry as we will only use loggers that are available for this test.